tools/poly_console.py

```
@@ -0,0 +1,107 @@
+            # truncates long lines and drops continuations (the same trap
+            # documented for polykybd-ctnd's perf_runner console handling).
+            buf += bytes(data).split(b"\x00", 1)[0].decode("utf-8", "replace")
+            while "\n" in buf:
+                line, buf = buf.split("\n", 1)
+                print(line.rstrip("\r"), flush=True)
+
+        except KeyboardInterrupt:
+            if buf:
+                print(buf, flush=True)   # don't swallow a trailing partial line
+            return 0
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)